language/move-binary-format/src/check_bounds.rs

```diff
@@ -146,9 +210,12 @@ impl<'a> BoundsChecker<'a> {
         &self,
         struct_instantiation: &StructDefInstantiation,
     ) -> PartialVMResult<()> {
-        check_bounds_impl(&self.module.struct_defs, struct_instantiation.def)?;
         check_bounds_impl(
-            &self.module.signatures,
+            &self.view.struct_defs().unwrap_or(&[]),
```

Comment:
maybe making something for `check_bounds_impl_opt` that just gives an out of bounds error on `None` might be a bit more grokkable. Took me a bit to think through this working